kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/SubtreeManager.kt

```diff
@@ -81,13 +84,24 @@ internal class SubtreeManager<StateT, OutputT : Any>(
    * is managing.
    */
   fun <T : Any> tickChildren(
-    selector: SelectBuilder<T?>,
-    handler: (WorkflowAction<StateT, OutputT>) -> T?
+    selector: SelectBuilder<OutputEnvelope<T>>,
+    handler: (WorkflowAction<StateT, OutputT>, Kind) -> OutputEnvelope<T>
   ) {
     for ((case, host) in hostLifetimeTracker.lifetimes) {
       host.tick(selector) { output ->
-        val componentUpdate = case.acceptChildOutput(output)
-        return@tick handler(componentUpdate)
+        if (output.output != null) {
+          val componentUpdate = case.acceptChildOutput(output.output)
+          // This workflow's child emitted a non-null output, this workflow's action was invoked,
+          // meaning we did update ourselves.
+          return@tick handler(
+              componentUpdate,
+              Kind.DidUpdate(Source.Subtree(case.id.name, output.debugInfo))
+          )
+        } else {
+          // The child didn't actually emit an output, which means this workflow has nothing to do,
+          // which means we will only report our child updating.
+          return@tick handler(noAction(), Kind.ChildDidUpdate(output.debugInfo))
```

Comment:
There should probably be some kdoc somewhere saying that if the kind is `ChildDidUpdate`, `noAction` is the only valid workflow action.

Reply:
Added a bunch of kdoc to this method.